src/CombFilter/CombFilterIf.cpp

```diff
@@ -0,0 +1,128 @@
+
+// standard headers
+
+// project headers
+#include "MUSI6106Config.h"
+
+#include "ErrorDef.h"
+#include "Util.h"
+
+#include "CombFilterIf.h"
+#include "CombFilter.h"
+
+static const char*  kCMyProjectBuildDate = __DATE__;
+
+
+CCombFilterIf::CCombFilterIf () :
+    m_bIsInitialized(false),
+    m_pCCombFilter(0),
+    m_fSampleRate(0)
+{
+    // this should never hurt
+    this->reset ();
+}
+
+
+CCombFilterIf::~CCombFilterIf ()
+{
+    this->reset ();
+}
+
+const int  CCombFilterIf::getVersion (const Version_t eVersionIdx)
+{
+    int iVersion = 0;
+
+    switch (eVersionIdx)
+    {
+    case kMajor:
+        iVersion    = MUSI6106_VERSION_MAJOR; 
+        break;
+    case kMinor:
+        iVersion    = MUSI6106_VERSION_MINOR; 
+        break;
+    case kPatch:
+        iVersion    = MUSI6106_VERSION_PATCH; 
+        break;
+    case kNumVersionInts:
+        iVersion    = -1;
+        break;
+    }
+
+    return iVersion;
+}
+const char*  CCombFilterIf::getBuildDate ()
+{
+    return kCMyProjectBuildDate;
+}
+
+Error_t CCombFilterIf::create (CCombFilterIf*& pCCombFilter)
+{
+    pCCombFilter = new CCombFilterIf();
+    if (pCCombFilter == nullptr) {
+        return Error_t::kMemError;
+    }
+    return Error_t::kNoError;
+}
+
+Error_t CCombFilterIf::destroy (CCombFilterIf*& pCCombFilter)
+{
+    delete pCCombFilter;
```

Comment:
Do you need to call reset() first?

Reply:
Yes, that's true. Otherwise the internal state of initialized will last and affect next comb filter interface instance.

Reply:
Just also realized this will cause memory leakage as the `m_pCCombFilter` is not released. Good point, thank you!